closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -319,7 +333,7 @@ int encrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
      * This method is for CipherCore to insert the remainder of its buffer
      * into ibuffer_enc before a encryptFinal(ByteBuffer, ByteBuffer) operation
      */
-    int encrypt(byte[] in, int inOfs, int len) {
+    synchronized int encrypt(byte[] in, int inOfs, int len) {
```

Comment:
Don't synchronize the whole method, just the portion after the call to `nullAndBoundsCheck`.

Reply:
Done.